### ISSUE TITLE
UI freeze when watcher cannot find folder

### ISSUE THREAD

**Issue by sudhanv09** [gitbutlerapp/gitbutler#3001]
The UI is completely unresponsive due to the added project no longer existing. 

Here is the complete log:

```
2024-03-03T22:05:51.656259Z  INFO gitbutler-app/src/bin.rs:50: starting app version=0.10.14 name=GitButler
2024-03-03T22:05:51.802274Z ERROR gitbutler-app/src/watcher.rs:72: watcher error error=Other(failed to run dispatcher

Caused by:
    0: failed to run file change dispatcher
    1: failed to start watcher
    2: No such file or directory (os error 2)) project_id=1c207676-dc79-4a15-8d42-e06d9a3c81b9
2024-03-03T22:05:51.803528Z ERROR gitbutler-app/src/watcher.rs:72: watcher error error=Other(failed to run dispatcher

Caused by:
    0: failed to run file change dispatcher
    1: failed to start watcher
    2: No such file or directory (os error 2)) project_id=1db1608f-3a04-4020-9765-c31cefdc11e2
2024-03-03T22:05:52.918810Z  INFO get_user: gitbutler-app/src/users/commands.rs:23: close time.busy=3.02ms time.idle=102µs
2024-03-03T22:05:52.922931Z  INFO list_projects: gitbutler-app/src/projects/commands.rs:128: close time.busy=50.9µs time.idle=85.1µs
2024-03-03T22:05:53.106185Z  INFO git_head: gitbutler-app/src/commands.rs:67: close time.busy=206µs time.idle=139µs project_id="1c207676-dc79-4a15-8d42-e06d9a3c81b9"
2024-03-03T22:05:53.117166Z  INFO menu_item_set_enabled: gitbutler-app/src/menu.rs:11: close time.busy=12.1µs time.idle=260µs menu_item_id="project/settings" enabled=true
2024-03-03T22:05:53.117438Z  INFO fetch_from_target: gitbutler-app/src/virtual_branches/commands.rs:575: close time.busy=1.45ms time.idle=124µs project_id="1c207676-dc79-4a15-8d42-e06d9a3c81b9"

** (git-butler:297116): WARNING **: 06:06:16.816: atk-bridge: get_device_events_reply: unknown signature
```

Reproduce:
1. Add a project to gitbutler 
2. Delete the folder

**Comment by mtsgrd** [comment]
Thanks @sudhanv09, we'll definitely want to fix.

**Comment by Byron** [comment]
Even though the behaviour around deleting a project is still to be improved (tracked in #3447 for now), I wasn't able to reproduce the UI freeze described here with the latest version from #3471.

https://github.com/gitbutlerapp/gitbutler/assets/63622/698b415d-31af-4f3c-a1e4-38dbb3d576fc

Thus I think this issue can be considered resolved once #3471 is merged and available in the a non-nightly release.

**Comment by Byron** [comment]
Closing, as the fix has been released and is available in 0.11.2 or newer.
